packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -1035,15 +1038,21 @@ export const ComboBoxMixin = (subclass) =>
 
     /** @private */
     _indexOfValue(value, items) {
-      if (items && this._isValidValue(value)) {
-        for (let i = 0; i < items.length; i++) {
-          if (items[i] !== this.__placeHolder && this._getItemValue(items[i]) === value) {
-            return i;
-          }
-        }
+      if (!items || !this._isValidValue(value)) {
+        return -1;
       }
 
-      return -1;
+      return items.findIndex((item) => {
+        if (item instanceof ComboBoxPlaceholder) {
+          return false;
+        }
```

Comment:
**note:** This fixes the regression discovered by the ComboBox ITs. Placeholders can be instantiated outside the web component (in the connector, for example), so it is not enough to check for `item === this.__placeHolder` how it was previously.